src/main/java/sparksoniq/semantics/visitor/RuntimeIteratorVisitor.java

```diff
@@ -81,34 +82,36 @@ public RuntimeIterator visitDescendants(ExpressionOrClause expression, RuntimeIt
     @Override
     public RuntimeIterator visitFlowrExpression(FlworExpression expression, RuntimeIterator argument) {
         FlworClause startClause = expression.getStartClause();
-        List<FlowrClauseSparkIterator> iterators = new ArrayList<>();
-        iterators.addAll(this.visitFlowrClause(startClause, argument));
+        List<RuntimeTupleIterator> iterators = new ArrayList<>();
+        RuntimeTupleIterator previous = null;
+        previous = this.visitFlowrClause(startClause, argument, previous);
+        iterators.add(previous);
         for (FlworClause clause : expression.get_contentClauses())
-            iterators.addAll(this.visitFlowrClause(clause, argument));
-        ReturnClauseSparkIterator returnIterator =
-                (ReturnClauseSparkIterator) (this.visitFlowrClause(expression.get_returnClause(), argument).get(0));
+            previous = this.visitFlowrClause(clause, argument, previous);
+            iterators.add(previous);
```

Comment:
Isn't a pair of curly braces missing around these two lines? (I think this list may actually be unused and will be dropped later on, but for the sake of correctness, we may want to be careful).